src/modules/popupModal.js

```diff
@@ -0,0 +1,88 @@
+import getDetail from './getDetail.js';
+import fetchComments from './fetchComments.js';
+import submitComment from './submitComment.js';
+import commentCounter from './commentCounter.js';
+
+const newModal = document.getElementById('new-modal');
+const newClosebtn = document.getElementById('new-close');
+const newItemSection = document.getElementById('new-itemSection');
+const newCommentSection = document.getElementById('new-commentSection');
+const newCommentsHeading = document.getElementById('new-commentsHeading');
+const newBtnAddComment = document.getElementById('new-btnAddComment');
+const newUsernameInput = document.getElementById('new-username');
+const newCommentInput = document.getElementById('new-comment');
+
+const clearElements = (elements) => {
+  elements.forEach((element) => {
+    element.innerHTML = '';
+  });
+};
+
+const closeModal = () => {
+  newModal.style.display = 'none';
+  clearElements([newItemSection, newCommentsHeading, newCommentSection]);
+};
+
+const displayItemDetail = async (id, data) => {
+  newModal.style.display = 'flex';
+
+  newItemSection.innerHTML = `
+    <img class='img-beautiful' src='${data.strMealThumb}'>
+    <h2>${data.strMeal}</h2>
+    <p class='p'>${data.strInstructions}</p>
+  `;
+
+  const fetchAndDisplayComments = async () => {
+    const comments = await fetchComments(`RjyF2atccOw1gRFQE3W0/comments?item_id=${id}`);
+    if (comments.length > 0) {
+      const count = commentCounter(comments);
+      newCommentsHeading.innerHTML = `<p class='commentCountTitle'>Comments(${count})</p>`;
+      newCommentSection.innerHTML = comments
+        .map(
+          (comment) => `
+            <div class="commentBox-beautiful">
+              <p class="comment-name">${comment.username}:</p>
+              <p class="commentText-beautiful">${comment.comment}</p>
+              <p class="date-beautiful">${comment.creation_date}</p>
+            </div>
+          `,
+        )
+        .join('');
+    }
+  };
```

Comment:
- [ ] As requested by the previous reviewers, please there are still some issues we need to fix 😊

- [ ] When the Comments popup loads, the web app is expected to retrieve data from the selected API and show details about the selected item. 

- [ ] And likewise, retrieve data from the [Involvement API](https://www.notion.so/Involvement-API-869e60b5ad104603aa6db59e08150270) to show the item comments.

- [ ] In your app, the comments are being fetched continuously and comments from other cards are also displayed.

- [ ] Kindly check the loom [video](https://www.loom.com/share/c8f985e008424b4d95bb82deb8f9f4e1?sid=8238f019-83d2-45f2-ac69-56189434eb4e) I have recorded on this.